File: testdata/jennies/rawtypes/constant_reference_discriminator/JavaRawTypes/constant_reference_discriminator/RowsLayoutWithoutValue.java
package constant_reference_discriminator;

import java.util.Objects;

public class RowsLayoutWithoutValue {
    public String kind;
    public String rowsLayoutProperty;
    public RowsLayoutWithoutValue() {
        this.kind = Constants.RowsLayoutKindType;
        this.rowsLayoutProperty = "";
    }
    public RowsLayoutWithoutValue(String rowsLayoutProperty) {
        this.kind = Constants.RowsLayoutKindType;
        this.rowsLayoutProperty = rowsLayoutProperty;
    }

    @Override
    public boolean equals(Object other) {
        if (this == other) return true;
        if (!(other instanceof RowsLayoutWithoutValue)) return false;
        RowsLayoutWithoutValue o = (RowsLayoutWithoutValue) other;
        if (!Objects.equals(this.kind, o.kind)) return false;
        if (!Objects.equals(this.rowsLayoutProperty, o.rowsLayoutProperty)) return false;
        return true;
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.kind, this.rowsLayoutProperty);
    }
}

File: testdata/jennies/rawtypes/constant_reference_discriminator/CSharpRawTypes/src/Grafana/Foundation/ConstantReferenceDiscriminator/RowsLayoutWithoutValue.cs
namespace Grafana.Foundation.ConstantReferenceDiscriminator;


public class RowsLayoutWithoutValue
{
    public string Kind;
    public string RowsLayoutProperty;

    public RowsLayoutWithoutValue()
    {
        this.Kind = default!;
        this.RowsLayoutProperty = "";
    }

    public RowsLayoutWithoutValue(string kind, string rowsLayoutProperty)
    {
        this.Kind = kind;
        this.RowsLayoutProperty = rowsLayoutProperty;
    }
}
